src/libServer/ScillaIPCServer.h

```diff
@@ -25,6 +25,15 @@
 
 #include "libData/AccountData/Account.h"
 
+// An abstract base class to provide various blockchain info to Scilla.
+class ScillaBCInfo {
```

Comment:
Just curious, is the abstraction because the possibility of different providers exists in the near future? Because if not, maybe we can consider removing the abstraction for now, if we want to squeeze out whatever optimization we can on this front. The code looks good, but the creation of virtual table for this class (with only a single function implementation for each table entry) adds to execution time.